content/blog/architecture-as-code-intro/index.md

```diff
@@ -0,0 +1,307 @@
+
+If virtual machines are part of your infrastructure, Pulumi lets you create virtual machines programmatically across cloud service providers. The Javascript example below illustrates how to deploy a virtual machine on Google Cloud Platform. The script spins up a Debian based instance and creates a network with an open port 22 to let you configure and manage the machine as needed.
+
+```js
+const gcp = require("@pulumi/gcp");
+const fs = require(‘fs’);
+
+// Create a network
+const network = new gcp.compute.Network("network");
+const computeFirewall = new gcp.compute.Firewall("firewall", {
+    network: network.id,
+    allows: [{
+        protocol: "tcp",
+        ports: [ "22",”80” ],
+    }],
+});
+
+// Read the configuration script
+const startupScript = fs.readFileSync(‘config.sh’, ‘utf-8’)
+
+// Create a Virtual Machine Instance
+const computeInstance = new gcp.compute.Instance("instance", {
+    machineType: "f1-micro",
+    zone: "us-central1-a",
+    metadataStartupScript: startupScript,
+    bootDisk: { initializeParams: { image: "debian-cloud/debian-9" } },
+    networkInterfaces: [{
+        network: network.id,
+        // accessConfigus must includ a single empty config to request an ephemeral IP
+        accessConfigs: [{}],
+    }],
+});
+
+// Export the name and IP address of the Instance
+exports.instanceName = computeInstance.name;
+exports.instanceIP = computeInstance.networkInterfaces.apply(ni => ni[0].accessConfigs[0].natIp);
+```
+
+This is the configuration script.
+
+```bash
+#!/bin/bash
+echo "Hello, World!" > index.html
+nohup python -m SimpleHTTPServer 80 &
+```
+
+When the virtual machine is created, it reads the configuration file that creates a webserver. This a basic example, but it shows how to configure virtual machines programmatically, which is useful for scaling horizontally.
+
+## Serverless
+
+Serverless is a computing architecture characterized by server-side logic running in stateless containers that are invoked by events. They are typically ephemeral (often are available for only one call), and managed by third-party cloud providers. Serverless is also called Functions as a Service, or `FaaS`, and well-known implementations include AWS Lambda and Fargate, Azure Functions, and Google Cloud Run and Cloud Functions.
+
+A significant benefit of serverless is that it's a polyglot platform that allows developers to choose languages optimized for a task. For example, scripting languages like Javascript or Python may be more responsive than a language such as Java. Serverless functions can support both synchronous and asynchronous calls. There are use cases that require an immediate response such as processing a video stream and instances where returning the result immediately is not critical as in an ETL batch job.
```

Comment:
Javascript => JavaScript (applies throughout)

```suggestion
A significant benefit of serverless is that it's a polyglot platform that allows developers to choose languages optimized for a task. For example, scripting languages like JavaScript or Python may be more responsive than a language such as Java. Serverless functions can support both synchronous and asynchronous calls. There are use cases that require an immediate response such as processing a video stream and instances where returning the result immediately is not critical as in an ETL batch job.
```

Reply:
> For example, scripting languages like Javascript or Python may be more responsive than a language such as Java.

Really? Just because startup time?